src/main/java/randoop/generation/ObjectPool.java

```diff
@@ -0,0 +1,206 @@
+package randoop.generation;
+
+import static randoop.util.EquivalenceChecker.equivalentTypes;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import randoop.DummyVisitor;
+import randoop.ExecutionOutcome;
+import randoop.NormalExecution;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.test.DummyCheckGenerator;
+import randoop.types.Type;
+import randoop.util.ListOfLists;
+import randoop.util.SimpleArrayList;
+import randoop.util.SimpleList;
+
+/**
+ * A class representing a pool of objects, each associated with a list of sequences. Used in
+ * randoop.generation.Detective.
+ */
+public class ObjectPool {
+  /** The underlying data structure storing the objects and their associated sequences */
+  private final LinkedHashMap<Object, SimpleList<Sequence>> objPool;
+
+  /** Default constructor that initializes the object pool. */
```

Comment:
A default constructor is one automatically inserted when a programmer writes no constructor.  So, this isn't a default constructor.